server/src/main/java/org/elasticsearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -83,6 +86,13 @@ public final SortField sortField(
         Nested nested,
         boolean reverse
     ) {
+        if (targetNumericType == NumericType.DATE_NANOSECONDS) {
+            // explicitly set missingValue if not set yet to prevent negative values that would cause invalid nanosecond range
+            if (sortMissingFirst(missingValue) || sortMissingLast(missingValue)) {
+                final boolean min = sortMissingFirst(missingValue) ^ reverse;
+                missingValue = min ? 0L : Long.MAX_VALUE;
+            }
+        }
```

Comment:
This logic could move to `IndexNumericFieldData#missingObject` ? `IndexFieldData#missingObject` doesn't need to be final, we need to pick the value depending on the `targetNumericType`.

Reply:
I looked into this a bit and one problem is that I couldn't solve with this is that "missingObject" is part of `XFieldComparatorSource`, not `IndexFieldData`. That probably woudl mean overwriting it in special versions of LongValuesComparatorSource, and for the things I tried it wasn't always clear how to ther the "targetNumericType" from the sort (e.g. reduceType() returns `SortField.Type` that doesn't even differentiate between Long, Data and Datenanos like `IndexNumericFieldData.NumericType` does). Maybe I'm missing something but from what I tried this current location seems less complex to me.

Reply:
The `targetNumericType` could be added to the `LongValuesSource` and checked in `missingObject` ? It's not really about complexity here, `missingObject` is called in several contexts so it needs a fix.